COMPONENT-API.md

```diff
@@ -1,376 +1,864 @@
-The component API borrows concepts and structure from frontend components, like those in [Vue.js](https://vuejs.org/) or [React](https://reactjs.org/).
+Pipedream publishes a build of the CLI for macOS and Linux (for `386`, `amd64`, `arm`, and `arm64` architectures). If you need to use the CLI on another OS or architecture, [please reach out](https://docs.pipedream.com/support/).
 
-Components [run on Pipedream infrastructure](#how-components-run).
+Once the CLI is installed, log into Pipedream:
 
-## Features
+```bash
+pd login
+```
 
-Component can:
+See the [CLI reference](https://docs.pipedream.com/cli/reference/) for detailed usage and examples beyond those covered below.
 
-- Accept input via [props](#props)
-- Define [methods](#methods).
-- [`require` any npm package](#using-npm-packages)
-- Spin up Pipedream infrastructure, like HTTP servers and scheduled jobs, by declaring the correct [interfaces](#interfaces)
-- Store and retrieve state using the [built-in key-value store](#servicedb)
-- [Emit data](#thisemit) you process within the component, allowing you to access it outside of Pipedream via API.
+## Contributing
 
-## Reference
+Deploy or contribute to curated open source components in Pipedream's Github repo. Or author, deploy and maintain your own via your standard CI/CD process.
 
-<!--ts-->
+# Component Lifecycle
 
-- [name](#name)
-  - [Name slugs](#name-slugs)
-- [version](#version)
-- [run method](#run-method)
-- [Referencing this](#referencing-this)
-  - [Referencing props](#referencing-props)
-  - [Referencing methods](#referencing-methods)
-- [Interfaces](#interfaces)
-  - [`$.interface.http`](#interfacehttp)
-    - [Methods on the HTTP interface](#methods-on-the-http-interface)
-  - [`$.interface.timer`](#interfacetimer)
-- [props](#props)
-  - [Example](#example)
-- [methods](#methods)
-- [`this.$emit`](#thisemit)
-  - [How to emit events](#how-to-emit-events)
-  - [Retrieving events programmatically](#retrieving-events-programmatically)
-- [`$.service.db`](#servicedb)
-  - [set method](#set-method)
-  - [get method](#get-method)
-- [Using npm packages](#using-npm-packages)
-- [How components run](#how-components-run)
+## Diagram
 
-<!--te-->
+![image-20200816223906765](./images/image-20200816223906765.png)
 
-## `name`
+## States
 
-The name of the component, a **string** which identifies components deployed to users' accounts.
+Pipedream components support `Activate()` and `Deactivate()` lifecycle hooks. The code for these hooks are defined within the component. Learn more about the [component structure](#component-structure) and [hook usage](#hooks).
 
-This name will show up in the Pipedream UI, in CLI output (for example, from `pd list` commands), etc.
+### Saved Component
 
-### Name slugs
+A saved component is non-instantiated component code that has previously been deployed to Pipedream. Each saved component has a unique saved component ID. Saved components cannot be invoked directly — they must first be depoyed.
```

Comment:
"they must first be depoyed" -> deployed

Reply:
ack